src/plugins/better-evaluation-view/index.ts

```diff
@@ -1,24 +1,64 @@
-import { Inserter, PluginController, Replacer } from "../PluginController";
+import { EvaluationContainerComponent } from "#components";
+import { ConstantListValueType, TypedConstantValue, ValueType } from "#globals";
+import { PluginController, Replacer } from "../PluginController";
 import "./better-evaluation-view.less";
 import { ColorEvaluation } from "./components/ColorEvaluation";
 import { ListEvaluation } from "./components/ListEvaluation";
 import { Config, configList } from "./config";
 
+type EvaluableConstantValue = TypedConstantValue<
+  ConstantListValueType | ValueType.RGBColor
+>;
+
 export default class BetterEvaluationView extends PluginController<Config> {
   static id = "better-evaluation-view" as const;
   static enabledByDefault = true;
   static config = configList;
 
-  listEvaluation(val: Parameters<typeof ListEvaluation>[0]): Inserter {
-    if (!this.settings.lists) return undefined;
-    return () => ListEvaluation(val);
+  evaluation(val: () => EvaluableConstantValue | undefined): Replacer {
+    const { settings } = this;
+    const value = val();
+    if (!value) return undefined;
+    switch (value.valueType) {
+      case ValueType.ListOfAngleMarker:
+      case ValueType.ListOfDirectedAngleMarker:
+      case ValueType.ListOfTransformation:
+      case ValueType.ListOfConfidenceInterval:
+      case ValueType.ListOfOneSampleTInference:
+      case ValueType.ListOfTwoSampleTInference:
+      case ValueType.ListOfRegressionTInference:
+      case ValueType.ListOfOneSampleZInference:
+      case ValueType.ListOfTwoSampleZInference:
+      case ValueType.ListOfOneProportionZInference:
+      case ValueType.ListOfTwoProportionZInference:
+      case ValueType.ListOfZSignificanceTest:
+      case ValueType.ListOfTSignificanceTest:
+      case ValueType.ListOfChiSquareGoodnessOfFit:
+      case ValueType.ListOfChiSquareIndependence:
+        return undefined;
+      case ValueType.RGBColor:
+        return settings.colors
+          ? (swatch) => ColorEvaluation(value, swatch)
+          : undefined;
+      case ValueType.ListOfColor:
+        return settings.colors && settings.lists && settings.colorLists
+          ? (swatch) => ColorEvaluation(value, swatch)
+          : undefined;
+      default:
+        return settings.lists ? () => ListEvaluation(value) : undefined;
```

Comment:
This makes me suspicious because it implies that the cases excluded above guarantee that `value` has type `NormalListValueType`. But if Desmos ever adds a new list type, then this assumption can break. In this case, I know it won't _crash_ because `ListEvaluation` only relies on the type in `formatLabels` which has a safe `default` case, but it'll still look weird.

But I'd rather it be the other way around: have a bunch of cases (`case ListOfNumber:`, `case ListOfComplex:` etc) that limit the type of `value` to a `NormalListValueType`, then let the `default` case be the `return undefined` that `ListOfChiSquareIndependence` etc should reach. Then it would work even if you never thought about `ListOfChiSquareIndependence` in the first place.